firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/analytics/AnalyticsConnectorReceiver.java

```diff
@@ -66,6 +67,29 @@ public boolean register() {
     analyticsConnectorHandle =
         analyticsConnector.registerAnalyticsConnectorListener(CRASHLYTICS_ORIGIN, this);
 
+    if (analyticsConnectorHandle == null) {
+      Logger.getLogger()
+          .d(
+              "Could not register AnalyticsConnectorListener with origin "
+                  + "\""
+                  + CRASHLYTICS_ORIGIN
+                  + "\"");
+      // Older versions of FA don't support CRASHLYTICS_ORIGIN. We can try using the old Firebase
+      // Crash Reporting origin
+      analyticsConnectorHandle =
+          analyticsConnector.registerAnalyticsConnectorListener(LEGACY_CRASH_ORIGIN, this);
+
+      // If FA allows us to connect with the legacy origin, but not the new one, nudge customers
+      // to update their FA version.
+      if (analyticsConnectorHandle != null) {
+        Logger.getLogger()
+            .w(
+                "Outdated version of Firebase Analytics detected. For improved "
```

Comment:
I'm going to coordinate with the Analytics team on the specific wording, which is likely to change. I created the PR to get feedback on the general approach; don't worry about the warning message just yet.